plugins/bazaar/src/components/AddProjectDialog/AddProjectDialog.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect, Dispatch, SetStateAction } from 'react';
+import { Entity } from '@backstage/catalog-model';
+import { SubmitHandler } from 'react-hook-form';
+import { updateMetadata } from '../../util/dbRequests';
+import { useApi, configApiRef } from '@backstage/core-plugin-api';
+import { ProjectDialog } from '../ProjectDialog';
+import { ProjectSelector } from '../ProjectSelector';
+import { BazaarProject, FormValues, Status } from '../../util/types';
+
+type Props = {
+  catalogEntities: Entity[];
+  open: boolean;
+  handleClose: () => void;
+  setBazaarProjects: Dispatch<SetStateAction<BazaarProject[]>>;
+  setCatalogEntities: Dispatch<SetStateAction<Entity[]>>;
+};
+
+export const AddProjectDialog = ({
+  catalogEntities,
+  open,
+  handleClose,
+  setBazaarProjects,
+  setCatalogEntities,
+}: Props) => {
+  const baseUrl = useApi(configApiRef)
```

Comment:
So I think what would be great here is for us to create a `bazaarApiRef` which would be used to get access to an implementation that would allow us to call the backend.

Basically all the things that live in `../../util/dbRequests` should live in this `apiRef` and then you can move all the logic for that into that implementation.

Look at this for inspiration and how we do things in the `scaffolder` https://github.com/backstage/backstage/blob/a111b43771a2a825802e4102930b818c4ba79337/plugins/scaffolder/src/api.ts#L94

Reply:
Maybe this documentation could help you too https://backstage.io/docs/api/utility-apis#defining-custom-utility-apis